web/panic_capture.go

```diff
@@ -0,0 +1,30 @@
+package web
+
+import (
+	"net/http"
+
+	"github.com/edaniels/golog"
+)
+
+// PanicCapture allows recovery during a request handler from panics. It prints a
+// formatted log to the underlying logger.
+type PanicCapture struct {
+	Logger golog.Logger
+}
+
+// Recover captures and prints the error if recover() has an error.
+func (p *PanicCapture) Recover(w http.ResponseWriter, r *http.Request) {
+	err := recover()
+	if err == nil {
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/plain")
+	w.WriteHeader(http.StatusInternalServerError)
+	_, err = w.Write([]byte("internal server error"))
```

Comment:
if recover has an error, we are losing it at this point? I assume you want to log it on the last line but now it's the .Write err instead

Reply:
Ah yeah that was a last minute add from lint :)